### PR TITLE
Cleanup (navmesh-related)

### PR DESCRIPTION
**Pull request by slipher**

### CODE REVIEW

File: src/sgame/sg_bot_parse.cpp [hunk 1]
```diff
@@ -1470,57 +1469,14 @@ void FreeTokenList( pc_token_list *list )
...
-		if ( !Q_stricmp( testTree->name, tree->name ) )
-		{
-			FreeBehaviorTree( testTree );
-			memmove( &list->trees[ i ], &list->trees[ i + 1 ], sizeof( AIBehaviorTree_t * ) * ( list->numTrees - i - 1 ) );
-			list->numTrees--;
-		}
-	}
-}
-
 void FreeTreeList( AITreeList_t *list )
```
Comment (DolceTriade) [inline src/sgame/sg_bot_parse.cpp:1472]: is a list->clear() required?

**Review by DolceTriade — APPROVED** [review]
```
/usr/bin/g++-8 -DARCH_STRING=amd64 -DBUILD_SGAME -DBUILD_VM -DBUILD_VM_IN_PROCESS -DDAEMON_ARCH_amd64 -DDAEMON_BUILD_Debug -DDEBUG_BUILD -DGLM_FORCE_EXPLICIT_CTOR -DNACL_ANDROID=0 -DNACL_ARCH_STRING=x86_64 -DNACL_BUILD_ARCH=x86 -DNACL_BUILD_SUBARCH=64 -DNACL_LINUX=1 -DNACL_OSX=0 -DNACL_WINDOWS=0 -DVM_NAME=sgame -Dsgame_native_dll_EXPORTS -I/home/vsts/work/1/s/daemon/src -I/home/vsts/work/1/s/daemon/libs -I/home/vsts/work/1/s/daemon/libs/zlib -I/home/vsts/work/1/s/daemon/libs/nacl -I/home/vsts/work/1/s/libs -I/home/vsts/work/1/s/src -I/usr/include/freetype2 -I/home/vsts/work/1/s/libs/RmlUi/Include -I/usr/include/lua5.2 -I/home/vsts/work/1/s/libs/recastnavigation/Detour/Include -I/home/vsts/work/1/s/libs/recastnavigation/DetourTileCache/Include -I/home/vsts/work/1/s/libs/recastnavigation/DebugUtils/Include -I/home/vsts/work/1/s/libs/recastnavigation/DetourCrowd/Include -I/home/vsts/work/1/s/libs/recastnavigation/Recast/Include -I/home/vsts/work/1/s/libs/recastnavigation/RecastDemo/Include -I/home/vsts/work/1/s/libs/glm -ffast-math -fno-strict-aliasing -m64 -mtune=generic -mcx16 -fvisibility=hidden -fno-gnu-unique -std=gnu++14 -pthread -g -g3 -fPIC -Wall -Wextra -Werror -Wno-pragmas -Wno-unknown-pragmas -Wno-error=old-style-cast -Woverloaded-virtual -Wstrict-null-sentinel -Wno-error=sign-promo -Wno-error=switch -Wno-error=unused-function -Wno-error=unused-parameter -Wno-maybe-uninitialized -MD -MT CMakeFiles/sgame-native-dll.dir/src/sgame/botlib/bot_load.cpp.o -MF CMakeFiles/sgame-native-dll.dir/src/sgame/botlib/bot_load.cpp.o.d -o CMakeFiles/sgame-native-dll.dir/src/sgame/botlib/bot_load.cpp.o -c /home/vsts/work/1/s/src/sgame/botlib/bot_load.cpp
/home/vsts/work/1/s/src/sgame/botlib/bot_load.cpp: In function ‘bool BotLoadNavMesh(const char*, NavData_t&)’:
/home/vsts/work/1/s/src/sgame/botlib/bot_load.cpp:214:42: error: base operand of ‘->’ has non-pointer type ‘NavData_t’
  BotLoadOffMeshConnections( filename, nav->process.con );
```

Build error looks legit

Allin all, looks good minus build error and comment.

File: src/sgame/sg_bot_parse.cpp (hunk 1 above)
Comment (slipher) [reply src/sgame/sg_bot_parse.cpp:1472]: Yes and there already is one.

Comment (DolceTriade) [reply src/sgame/sg_bot_parse.cpp:1472]: oops must have missed it in the diff.

**Review by DolceTriade — APPROVED** [review]

**Pull request merged**